src/platform/Zephyr/DeviceInfoProviderImpl.cpp

```diff
@@ -0,0 +1,366 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/core/CHIPTLV.h>
+#include <lib/support/CHIPMemString.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/DefaultStorageKeyAllocator.h>
+#include <platform/Zephyr/DeviceInfoProviderImpl.h>
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <cstring>
+
+namespace chip {
+namespace DeviceLayer {
+
+namespace {
+constexpr TLV::Tag kLabelNameTag  = TLV::ContextTag(0);
+constexpr TLV::Tag kLabelValueTag = TLV::ContextTag(1);
+} // anonymous namespace
+
+DeviceInfoProviderImpl & DeviceInfoProviderImpl::GetDefaultInstance()
+{
+    static DeviceInfoProviderImpl sInstance;
+    return sInstance;
+}
+
+DeviceInfoProvider::FixedLabelIterator * DeviceInfoProviderImpl::IterateFixedLabel(EndpointId endpoint)
+{
+    return chip::Platform::New<FixedLabelIteratorImpl>(endpoint);
+}
+
+DeviceInfoProviderImpl::FixedLabelIteratorImpl::FixedLabelIteratorImpl(EndpointId endpoint) : mEndpoint(endpoint)
+{
+    mIndex = 0;
+}
+
+size_t DeviceInfoProviderImpl::FixedLabelIteratorImpl::Count()
+{
+    // In Zephyr Simulation, return the size of the hardcoded labelList on all endpoints.
```

Comment:
What is "Zephyr Simulation"?  How would an actual device maker implement an actual fixed label cluster on top of this code?

Reply:
It is basically a copy & paste from other platforms - I see amebad and linux being basically the same thing.

As I understand it, each application needs to implement this, however the paths to bind this at application level seems to not be implemented and this exists rather at the platform level.

If we are to re-make the bindings, I believe this should happen as a separate PR/effort. Right now this is "do the same as linux/darwin/ameba".

Reply:
I know it's a copy/paste.  My point is that for Linux and Darwin this approach maybe made sense in the "this is just test code, no one is going to actually use it" sense, but in general this approach does not make sense, and checking it in just gives us a false sense of "it's implemented" when it's actually not.

Let me turn the question around: what is the point of adding this code?  As far as I can tell, the only point is to make it looks like tests pass when they would not actually do that for a real application.  I don't think we should be going there.

Reply:
Chatted offline, I will move the existing implementation (which seems to be copied and pasted across implementations) to a central place that can be reused by various app main.

Reply:
Updated to a common implementation.